stdlib/in_channel.mli

```diff
@@ -0,0 +1,164 @@
+
+val input_byte : t -> int option
+(** Same as {!input_char}, but return the 8-bit integer representing
+    the character.
+    Return [None] if an end of file was reached. *)
+
+val input_binary_int : t -> int option
+(** Read an integer encoded in binary format (4 bytes, big-endian)
+    from the given input channel. See {!Stdlib.output_binary_int}.
+    Return [None] if an end of file was reached while reading the
+    integer. *)
+
+val input_line : t -> string option
+(** Read characters from the given input channel, until a
+    newline character is encountered. Return the string of
+    all characters read, without the newline character at the end.
+    Return [None] if the end of the file is reached
+    at the beginning of line. *)
+
+val input_lines : t -> string list
+(** TBA *)
+
+val iter_lines : (string -> unit) -> t -> unit
+(** TBA *)
+
+val fold_lines : ('a -> string -> 'a) -> 'a -> t -> 'a
+(** TBA *)
+
+val input_all : t -> string
+(** TBA *)
+
+val seek : t -> int64 -> unit
+(** [seek chan pos] sets the current reading position to [pos] for channel
+    [chan]. This works only for regular files. On files of other kinds, the
+    behavior is unspecified. *)
+
+val pos : t -> int64
+(** Return the current reading position for the given channel. *)
+
+val length : t -> int64
+(** Return the size (number of characters) of the regular file
+    on which the given channel is opened.  If the channel is opened
+    on a file that is not a regular file, the result is meaningless.
+    The returned size does not take into account the end-of-line
+    translations that can be performed when reading from a channel
+    opened in text mode. *)
+
+val set_binary_mode : t -> bool -> unit
+(** [set_binary_mode ic true] sets the channel [ic] to binary
+    mode: no translations take place during input.
+    [set_binary_mode ic false] sets the channel [ic] to text
+    mode: depending on the operating system, some translations
+    may take place during input.  For instance, under Windows,
+    end-of-lines will be translated from [\r\n] to [\n].
+    This function has no effect under operating systems that
+    do not distinguish between text mode and binary mode. *)
+
+(** {2 Input functions on filenames} *)
+
+val read_lines : string -> binary:bool -> string list
+(** TBA *)
+
+val read_all : string -> binary:bool -> string
+(** TBA *)
+
+(** {2 Input functions on standard input} *)
+
+val read_float : unit -> float option
+(** Flush standard output, then read one line from standard input
+    and convert it to a floating-point number.
+
+    Return [None] if the line read is not a valid representation of a
+    floating-point number.
+*)
+
+val read_int : unit -> int option
+(** Flush standard output, then read one line from standard input
+    and convert it to an integer.
+
+    Return [None] if the line read is not a valid representation of an integer.
+*)
+
+val read_line : unit -> string option
```

Comment:
It is confusing to have both `read_line : unit -> string option` and `read_lines : string -> binary:bool -> string list`. Not sure what is the best way to handle this? 1) rename one of them, 2) omit one of them, 3) ...

Reply:
I agree. There's nothing to signify that `read_lines` concerns a file -- you have to go in and read the documentation. `read_file_lines` and `read_file_all` or `file_read_lines` and `file_read_all` would make more sense to me.

Reply:
Can we make `read_*` take `t` rather than `unit`?  It's slightly more verbose when reading from stdin, but also explicit about where the data are coming from.

Or they could be named `read_float_stdin`, etc.

Reply:
@hcarty The functions taking `t` as argument are already there with prefix `input_*`. Another possibility is to put all the functions specialised to `stdin`, `stdout` and `stderr` in their own module, say `Stdio`.